include/rapidjson/allocators.h

```diff
@@ -178,9 +178,11 @@ class MemoryPoolAllocator {
             return NULL;
 
         size = RAPIDJSON_ALIGN(size);
-        if (chunkHead_ == 0 || chunkHead_->size + size > chunkHead_->capacity)
-            if (!AddChunk(chunk_capacity_ > size ? chunk_capacity_ : size))
+        if (chunkHead_ == 0 || chunkHead_->size + size > chunkHead_->capacity) {
+            size = chunk_capacity_ > size ? chunk_capacity_ : size;
```

Comment:
This size will be assigned later, we need to ensure the correct value is used in the allocation case.